### DISCUSSION TITLE
Managing Detections with Missing Audio Files

### DISCUSSION THREAD

**Discussion by Vollpflock** [Mattk70/Chirpity-Electron#106] [General]
I renamed and relocated some folders. Since https://github.com/Mattk70/Chirpity-Electron/issues/14 is still an issue, my archive is full of detections without audio. That means I get the following message a lot when scrolling through archive: "Unable to locate source file with any supported file extension: blabla"

Is there a way to delete all of these detections?

**Comment by Mattk70** [comment]
Hmm, yes, I should really give this a bit more attention. I did have a prompt to delete the record when the message popped up, but I kept clicking 'delete' when the reason the file wasn't found was that my NAS was offline. 

I can think of 3 options you might want to select: 
1. 'ignore' : because e.g. your NAS ia offline or SD card isn't inserted
2. 'delete': because you ditched the file. I *think*  this should delete all that file's records too (i.e. the same result as selecting  the 'remove current file from archive' option in the Records menu)
3. 'find file': because you moved or renameed the file.  Once found and confirmed, this should update the file name in the database, and update all the records linking to the file. 

Ideally, there'd also be a 4th option to:
4. Update ALL file locations. That is, replace all the files in the database stored within the /foo/bar folder, to point to the /new/foobar folder.

**Reply by Mattk70** [reply]
Finally, I have a solution for 1 - 3. 

In 1.11.0, if the file is not found, you'll get the notification with two options: "Locate File", "Delete". If you click "Locate file" a dialog will open allowing you to pick the new file - whether it be in a new location, or you've either renamed it or converted it to another format. There are a couple of checks, to help ensure you've selected the correct update and notifications given if these fail:

1. If the file you picked is already in the database (to prevent duplicate files)
2. If the file you picked has a different duration to the original.

There are other ways to mess up, for example you choose the wrong file to replace but it has the same length as the original, but I'm hoping this is a good-enough start!

Once you refresh the results (e.g. unfilter, filter) the audio will be found and things will work again!
